src/lib/utils/piano.ts

```diff
@@ -0,0 +1,60 @@
+import type { PianoKeyNote } from "../../models/jam"
+
+const octaveNotes: PianoKeyNote[] = [
+  { midi: 21, name: ["A", "La"], black: false },
+  { midi: 22, name: ["A#", "La#"], black: true },
+  { midi: 23, name: ["B", "Ti"], black: false },
+  { midi: 24, name: ["C", "Do"], black: false },
+  { midi: 25, name: ["C#", "Do#"], black: true },
+  { midi: 26, name: ["D", "Re"], black: false },
+  { midi: 27, name: ["D#", "Re#"], black: true },
+  { midi: 28, name: ["E", "Mi"], black: false },
+  { midi: 29, name: ["F", "Fa"], black: false },
+  { midi: 30, name: ["F#", "Fa#"], black: true },
+  { midi: 31, name: ["G", "So"], black: false },
+  { midi: 32, name: ["G#", "So#"], black: true },
+]
+
+export function genPianoKeys(size: 49 | 61 | 76 | 88): PianoKeyNote[][] {
+  let notes: PianoKeyNote[] = []
+  let startIdx: number
+  let firstNoteMIDI: number
+
+  switch (size) {
+    case 49:
+      startIdx = 3
+      break
+    case 61:
+      startIdx = 3
+      break
+    case 76:
+      startIdx = 7
+      break
+    case 88:
+      startIdx = 0
+      break
+    default:
+      startIdx = 3
+  }
+
+  firstNoteMIDI = 21 + startIdx
```

Comment:
Maybe we should label 21 so it doesn't appear to be a magic number
```suggestion
  const firstNoteMIDI = 21
  const firstKeyboardNoteMIDI = 21 + startIdx
```

Side note, IMO, it's better practice to declare and initialize variables with `const`  (whenever possible) closer to where they are used, rather than declare all the variables at the top of the file and assign them later.

It leads the author to write more immutable values, leading to fewer bugs. It also keeps variables scoped to only where they're needed